### ISSUE TITLE
isTimedout never reports true

### ISSUE THREAD

**Issue by JCB-K** [splitio/react-client#4]
I'm trying to send an event to an internal tracking API whenever the Split SDK client times out, but I'm having no luck using the isTimedout flag.

I've deployed a little reusable test case right here: https://splitio-timeout-hciu0go4e.now.sh/. In the app, either:

- paste in a fake API key
- or block the Split.io domain in devtools and paste a real API key

The client will timeout, and `client.once(client.Event.SDK_READY_TIMED_OUT)` will be triggered, but the isTimedout flag never turns true. I've been fiddling around a bit with the `updateOnSdkTimedout` flag, but the docs aren't super clear as to where to input it... I've tried to put it on config and as a prop directly on SplitFactory, but neither works.

Here is the code for this example (application here: https://github.com/JCB-K/splitio-timeout):

```js
import {
  useClient,
  withSplitTreatments,
  SplitFactory
} from "@splitsoftware/splitio-react";

const Hooks = () => {
  const client = useClient();
  const { isReady, isTimedout } = client;
  const [timeoutCallBackCalled, setTimedout] = React.useState();

  React.useEffect(() => {
    console.log("split.io: adding client listener");
    client.once(client.Event.SDK_READY_TIMED_OUT, function() {
      setTimedout(true);
    });
  }, [client]);

  return (
    <>
      <h2>Hooks</h2>
      <pre>{JSON.stringify({ isReady, isTimedout })}</pre>
      <h2>Client</h2>
      <pre>{JSON.stringify({ "client.isTimedout": client.isTimedout })}</pre>
      <h2>Client.once</h2>
      <pre>{JSON.stringify({ timeoutCallBackCalled })}</pre>
    </>
  );
};

const Hoc = withSplitTreatments(["abc"])(({ isReady, isTimedout }) => {
  return (
    <>
      <h2>Hoc</h2>
      <pre>{JSON.stringify({ isReady, isTimedout })}</pre>
    </>
  );
});

export default function App() {
  const [apiKey, setApiKey] = React.useState("");
  const [ready, setReady] = React.useState(false);
  return (
    <>
      <input
        placeholder="paste API key here"
        onChange={e => setApiKey(e.target.value)}
        value={apiKey}
      />
      <button type="submit" onClick={() => setReady(true)}>
        Submit
      </button>
      {ready ? (
        <SplitFactory
          config={{
            core: {
              authorizationKey: apiKey,
              key: "somekey"
            },
            startup: {
              readyTimeout: 5 // 5 seconds
            }
            // updateOnSdkTimedout: true
          }}
          //updateOnSdkTimedout={true}
        >
          <Hooks />
          <Hoc />
        </SplitFactory>
      ) : (
        <p>Split Key not loaded</p>
      )}
    </>
  );
}
```

**Comment by chillaq** [comment]
Hi, sorry for the delay responding, we are pushing a fix for this issue this week.

Thanks
Bilal

**Comment by EmilianoSanchez** [comment]
Hi @JCB-K , 

We found an issue as you mentioned, due to how ES6 promises are implemented on different environments, leading to some scenarios where the SDK_READY_TIMED_OUT event was interpreted as a SDK_READY event. We are working on a release for this week.

Please keep in mind the following points:
- The React SDK is designed for client side, and not meant for server-side rendering. You shouldn't use it with Next.js or other React-based SSR framework. Instead, consider using the [Node SDK](https://help.split.io/hc/en-us/articles/360020564931-Node-js-SDK) or even the [Redux-SDK](https://help.split.io/hc/en-us/articles/360038851551-Redux-SDK) if you are using Redux in your project. 
- The line `const { isReady, isTimedout } = client;` in your sample is using internal variables that are not considered part of the public API, and may break without notice. To mitigate the need of accessing the readiness state of Split client when using React hooks, we will expose the Split context in our next release, so that users can access these values as follows: `const { isReady, isTimedout } = React.useContext(SplitContext););`.
- The correct way for setting the `updateOnSdkTimedout` flag is as a prop of `SplitFactory` (the second commented line), not a property of the config. We will review our documentation to make sure that these parameters are properly and clearly documented.

Thanks,
Emiliano

**Comment by JCB-K** [comment]
Great to hear a fix is coming up. 

- I mostly used next.js as a quick way to deploy the example code, we don't use it in production. We probably will be in the near future, so we'd definitely be interested in support for it.

- OK, good to know. Since both the HOC and and Render Prop versions of the API use `isReady` and `isTimedout` it's easy to believe that these are usable directly in the Hooks API as well.

**Comment by EmilianoSanchez** [comment]
@JCB-K 

We have released the version 1.1.0 of our React SDK. You can check the new doc details about `SplitContext` [here](https://help.split.io/hc/en-us/articles/360038825091-React-SDK#advanced-subscribe-to-events-and-changes), and also the updated [example](https://github.com/splitio/react-sdk-examples).

Thanks for your feedback :)
